uber/site_sections/registration.py

```diff
@@ -306,48 +298,79 @@ def undo_sale(self, session, id):
         session.delete(session.sale(id))
         return 'Sale deleted'
 
+    def check_in_form(self, session, id):
+        attendee = session.attendee(id)
+        return {
+            'attendee': attendee,
+            'groups': [
+                (group.id, (group.name if len(group.name) < 30 else '{}...'.format(group.name[:27], '...'))
+                         + (' ({})'.format(group.leader.full_name) if group.leader else ''))
+                for group in session.query(Group)
+                                    .options(joinedload(Group.leader))
+                                    .filter(Group.status != c.WAITLISTED,
+                                            Group.id.in_(
+                                                session.query(Attendee.group_id)
+                                                       .filter(Attendee.group_id != None, Attendee.first_name == '')
+                                                       .distinct().subquery()))
+                                    .order_by(Group.name)
+            ] if attendee.paid == c.PAID_BY_GROUP and not attendee.group_id else []
+        }
```

Comment:
This is just the query we were previously doing in the ``index`` page.  As you can see, we omit the query if we're not going to need it.